File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max=0;
        for(int i=0;i<s.length();i++){
            boolean [] seen=new boolean[128];
            for(int j=i;j<s.length();j++){
                if(seen[s.charAt(j)]){
                    break;
                }
                seen[s.charAt(j)]= true;
                max=Math.max(max,j-i+1);
            }
        }
        return max;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        seen=set()
        left=0
        ans=0

        for i in range (len(s)):
            while(s[i]in seen):
                seen.remove(s[left])
                left+=1
            seen.add(s[i])
            ans=max(ans,i-left+1)
        return ans                
                